src/Entity/Unit.php

```diff
@@ -382,4 +385,33 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) :
     return $fields;
   }
 
+  /**
+   * Get comma separated list of provided language names.
+   *
+   * @return string
+   *   Comma separated list of language names.
+   */
+  public function getProvidedLanguageNames(): string {
+    $provided_languages = $this->get('provided_languages');
+    if ($provided_languages->isEmpty()) {
+      return '';
+    }
+
+    $all_languages = $this->languageManager()->getStandardLanguageList();
+
+    $language_names = [];
+    foreach ($provided_languages as $provided_language) {
+      $langcode = $provided_language->getValue()['value'];
+      if (!isset($all_languages[$langcode])) {
+        continue;
+      }
+
+      $language = $all_languages[$langcode][0];
+      // phpcs:ignore
+      $language_names[] = $this->t($language);
```

Comment:
AFAIK `$language->getName()` should return the translated name. See `LanguageInterface`:
```php
  /**
   * Gets the name of the language.
   *
   * @return string
   *   The human-readable name of the language (in the language that was
   *   used to construct this object).
   */
  public function getName();
```